proknow-sdk-test/LogsTest/AuditTest.cs

```diff
@@ -0,0 +1,137 @@
+            Assert.AreEqual(patientItem.Method, "GET");
+            Assert.AreEqual(patientItem.PatientId, $"{_patientTwo.Id}");
+            Assert.AreEqual(patientItem.PatientMrn, "123457-Mrn");
+            Assert.AreEqual(patientItem.PatientName, "123457-Name");
+            Assert.AreEqual(patientItem.ResourceId, $"{_patientTwo.Id}");
+            Assert.AreEqual(patientItem.ResourceName, "123457-Name");
+            Assert.AreEqual(patientItem.StatusCode, "200");
+            Assert.AreEqual(patientItem.Uri, $"/workspaces/{_workspaceItemTwo.Id}/patients/{_patientTwo.Id}");
+            Assert.AreEqual(patientItem.UserName, userItem.Name);
+            Assert.AreEqual(patientItem.WorkspaceId, $"{_workspaceItemTwo.Id}");
+            Assert.AreEqual(patientItem.WorkspaceName, $"{_workspaceItemTwo.Name}");
+
+            await receivedAuditLogItem.Next();
+            receivedAuditLogItem = await receivedAuditLogItem.Next();
+
+            Assert.AreEqual(receivedAuditLogItem.Total, (uint)4);
+
+            var nextPatientItem = receivedAuditLogItem.Items[0];
+
+            Assert.AreEqual(nextPatientItem.Classification, "HTTP");
+            Assert.AreEqual(nextPatientItem.Method, "POST");
+            Assert.AreEqual(nextPatientItem.PatientId, $"{_patientTwo.Id}");
+            Assert.AreEqual(nextPatientItem.PatientMrn, "123457-Mrn");
+            Assert.AreEqual(nextPatientItem.PatientName, "123457-Name");
+            Assert.AreEqual(nextPatientItem.ResourceId, $"{_patientTwo.Id}");
+            Assert.AreEqual(nextPatientItem.ResourceName, "123457-Name");
+            Assert.AreEqual(nextPatientItem.StatusCode, "200");
+            Assert.AreEqual(nextPatientItem.Uri, $"/workspaces/{_workspaceItemTwo.Id}/patients");
+            Assert.AreEqual(nextPatientItem.UserName, userItem.Name);
+            Assert.AreEqual(nextPatientItem.WorkspaceId, $"{_workspaceItemTwo.Id}");
+            Assert.AreEqual(nextPatientItem.WorkspaceName, $"{_workspaceItemTwo.Name}");
+        }
+
+        [TestMethod]
+        public async Task FilterParametersTest()
+        {
+            FilterParameters filterParams = new FilterParameters();
+            filterParams.Types = new string[] { "patient_created" };
+            filterParams.PageSize = 1;
+            filterParams.StartTime = DateTime.Now.AddDays(-1); 
+            filterParams.EndTime = DateTime.Now;
+            filterParams.UserName ="Admin";
+            filterParams.PatientName = "2-Name";
+            filterParams.Classification = "HTTP";
+            filterParams.Methods = new string[] { "POST" };
+            filterParams.URI = $"/workspaces/1234/patients";
+            filterParams.UserAgent = "1234";
+            filterParams.IpAddress = "127.0.0.1";
+            filterParams.StatusCodes = new string[] { "200" };
+            filterParams.WorkspaceId = "eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee";
+            filterParams.ResourceId = "eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee";
+
+            try
```

Comment:
We are really testing to make sure the Invalid parameters exception is NOT thrown, so if there is an invalid parameter there is an exception thrown and we want the test failed

Reply:
The try-catch is not necessary because the test will fail if an exception is thrown.  The only time a try-catch is necessary in a test is if you are testing that a particular exception is thrown under certain conditions and you want to catch it to verify the exception.